### PR TITLE
Refactor yeoman generator Intro and Confirmation + add tests

### PR DESCRIPTION
**Pull request by rhasselle-oddball**
- refactor intro/confirmation page
- add some base tests
- ask question about missing vets-json-schema

Example of testing `yarn new:app` with these changes: https://github.com/department-of-veterans-affairs/vets-website/pull/31630

image
image
image
image
image
image

### CODE REVIEW

File: generators/form/templates/IntroductionPage.jsx.ejs [hunk 1]
```diff
@@ -1,99 +1,112 @@
...
+    <article className="schemaform-intro">
+      <FormTitle title={TITLE} subTitle={SUBTITLE} />
+      <h2 className="vads-u-font-size--h3 vad-u-margin-top--0">
+        Follow the steps below to apply for <%= benefitDescription %>.
+      </h2>
+      <ProcessList />
+      {showVerifyIdentify ? (
+        <div>{/* add verify identity alert if applicable */}</div>
+      ) : (
         <SaveInProgressIntro
```
Comment (Mottie) [inline generators/form/templates/IntroductionPage.jsx.ejs:77]: There's a [new intro page pattern for applications](https://design.va.gov/templates/forms/introduction#examples-application):

## Unauthenticated

Save in progress component _only below_ the process list

## Authenticated

- Verify alert (if applicable) _above_ the process list (not in design, but implied)
- Save in progress component _above_ and _below_ (button only) the process list

File: generators/form/index.js [hunk 2]
```diff
@@ -168,5 +171,95 @@ module.exports = class extends Generator {
...
+    const filePath = './src/platform/forms/constants.js';
+
+    if (this.props.formNumber && !this.props.usesVetsJsonSchema) {
+      const content = this.fs.read(filePath);
+
+      const newFormId = `FORM_${this.props.formNumber.replace(/-/g, '_')}`;
+
+      try {
+        const updatedContent = content.replace(
+          /(export const VA_FORM_IDS = Object\.freeze\({)([\s\S]*?)(}\))/,
```
Comment (Mottie) [inline generators/form/index.js:219]: There are a bunch more entries in the `constants.js` file. I'm not sure if we need to update them all?

- `getAllFormLinks`
- `TRACKING_PREFIXES`
- `SIP_ENABLED_FORMS` - do we even ask if the form will use SiP?
- `MY_VA_SIP_FORMS` - WIP maintained by the Authenticated Experience team

File: generators/form/templates/tests/fixtures/mocks/local-mock-responses.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// yarn mock-api --responses ./src/applications/{application}/tests/e2e/fixtures/mocks/local-mock-responses.js
+const mockUser = require('./user.json');
+
+const responses = {
+  'GET /v0/user': mockUser,
+};
+
+module.exports = responses;
```
Comment (Mottie) [inline generators/form/templates/tests/fixtures/mocks/local-mock-responses.js:8]: Awesome that you included this! 🎉

File: generators/form/templates/IntroductionPage.jsx.ejs (hunk 1 above)
Comment (rhasselle-oddball) [reply generators/form/templates/IntroductionPage.jsx.ejs:77]: I think we're going to be adjusting this area some more over the next couple of weeks with new DST changes to introduction page SIP/verify alert. I've also added a new ticket [[yeoman] unauthenticated vs authenticated question ](https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team/new?issueType=issue). Do you think it's fine to leave this as is for now and continue to adjust as followups?

File: generators/form/index.js (hunk 2 above)
Comment (rhasselle-oddball) [reply generators/form/index.js:219]: I've added this ticket as a followup to look into these [[yeoman] update other form id options in the constants file for things like SIP ](https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1593)

**Review by Mottie — APPROVED** [review]

**Review by ndsprinkle — APPROVED** [review]

**Review by jamigibbs — APPROVED** [review]

File: generators/form/templates/ConfirmationPage.jsx.ejs [hunk 4]
```diff
@@ -1,65 +1,82 @@
...
+    <div>
+      <div className="print-only">
+        <img
+          src="https://www.va.gov/img/design/logo/logo-black-and-white.png"
+          alt="VA logo"
+          width="300"
+        />
+      </div>
+
+      <va-alert uswds status="success" ref={alertRef}>
```
[the remark this answers is not in the document]
Comment (rhasselle-oddball) [reply generators/form/templates/ConfirmationPage.jsx.ejs:35]: good catch - I'm continuing to make followups, I will include this in the next PR.

File: generators/form/templates/IntroductionPage.jsx.ejs [hunk 5]
```diff
@@ -1,99 +1,112 @@
...
-          buttonOnly
           headingLevel={2}
           prefillEnabled={formConfig.prefillEnabled}
           messages={formConfig.savedFormMessages}
           pageList={pageList}
-          startText="Start the Application"
+          startText="Start the application"
+          // hideUnauthedStartLink
```
[the remark this answers is not in the document]
Comment (rhasselle-oddball) [reply generators/form/templates/IntroductionPage.jsx.ejs:83]: I intentionally left that there so the user sees it as an option - but maybe it is best to remove - but actually there is a [followup ticket to ask question unauthenticated or not](https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1592) - so I can add that as a task to that ticket.

**Pull request merged**